### ISSUE TITLE
[Bug] Pagination doesnt appear to be working correctly for result sets above 1000

### ISSUE THREAD

**Issue by BARNZ** [DirectoryTree/LdapRecord#315]
**Environment:**
 - LDAP Server Type: ActiveDirectory 
 - PHP Version: PHP 7.4.10

After updating this library to latest (`v2.5.3`) from the version we were previously on: `v1.12.6`, I've run into a weird bug with `paginate` on a large result set of users in our AD (2100+). The code below illustrates the problem:

Example code:
```php
$query = User::andFilter(function ($q) {
    // Some arbitrary filters for our own use-case
    $q
        ->where('extensionattribute11', 'PERSONAL')
        ->whereHas('mail')
        ->whereHas('givenname')
        ->whereHas('sn')
        ->where('c', 'NZ');
});

$count = 0;
foreach ($query->paginate(100) as $result) {
    $count++;
}
dd($count);
```

On `v1.12.6` this prints 2136 as expected, but on the latest version this same code only prints 1000 ?

Any idea what is going on here? I think its a bug, but I'm wondering why no one else has been tripped up by this yet.

No matter what parameter I use for the paginate call, I still get just 1000 results.

**Comment by stevebauman** [comment]
Hi @BARNZ,

Can you downgrade to `v2.4.*` and see if you still have the same issue?

**Comment by BARNZ** [comment]
Hi @stevebauman 

Have tried downgrading to `v2.4.8` and its working as expected - 2136 results

Have also tried a bunch of versions following that:

- `v2.5.0` works fine
- `v2.5.1` onwards triggers the problem

**Comment by stevebauman** [comment]
Thanks for the quick reply @BARNZ!

Are you sure you positive your environment is running PHP 7.4 and not < PHP 7.3? [This change](https://github.com/DirectoryTree/LdapRecord/commit/6c9e8a1275165a3da46bdf20f1015310098a968e) would have affected this, if your composer installation is pointed to a higher version of PHP than your actual web server.

Try pasting this into your `routes/web.php` and then visiting your application:

```php
dd(phpversion());
```

If your versions are indeed the same, we can troubleshoot from there 👍

**Comment by BARNZ** [comment]
Yep definitely 7.4

![image](https://user-images.githubusercontent.com/2202345/125328569-976d7800-e398-11eb-8db2-24fb5f72a36f.png)

**Comment by stevebauman** [comment]
That's still running in console though. You will have to let your web server handle the request to display its version.

**Comment by BARNZ** [comment]
I'm only working via the cli for this stuff. Regardless here's the same output dumped in my browser via the webserver:

![image](https://user-images.githubusercontent.com/2202345/125329595-cb956880-e399-11eb-9a16-31ab4d8cc765.png)

Can confirm there is no other version of php installed on this machine.

Cheers

**Comment by stevebauman** [comment]
Can you `composer require` the `dev-master` branch and try again? I've just pushed a commit to see if it's possibly the empty server controls being included in the search request.

**Comment by BARNZ** [comment]
Yep that looks like it worked - got 2134 results instead of the 1000

![image](https://user-images.githubusercontent.com/2202345/125331632-59725300-e39c-11eb-8b96-6eeed43306ea.png)

**Comment by stevebauman** [comment]
Ok perfect, thanks for working through this with me @BARNZ!

I'll create a new release with this patch included in just a moment 👍

**Comment by BARNZ** [comment]
Awesome thanks very much @stevebauman!

**Comment by stevebauman** [comment]
Happy to help @BARNZ! Okay, I've just released [v2.5.4](https://github.com/DirectoryTree/LdapRecord/releases/tag/v2.5.4) with this patch included 👍  You're all set!
